contracts/voting-verifier/src/state.rs

```diff
@@ -44,31 +46,28 @@ impl Poll {
 }
 
 #[cw_serde]
-pub struct PollMessage {
-    pub msg: Message,
-    pub poll_id: PollID,
+pub struct PollContent<T> {
+    pub content: T, // content is stored for migration purposes in case the hash changes
+    pub poll_id: PollId,
     pub index_in_poll: u32,
 }
 
-impl PollMessage {
-    pub fn new(msg: Message, poll_id: PollID, index_in_poll: usize) -> Self {
+impl<T> PollContent<T> {
+    pub fn new(content: T, poll_id: PollId, index_in_poll: usize) -> Self {
```

Comment:
this just moved the previously mentioned problem from the `verification_status` call to this constructor. I suggest creating two constructors, e.g. `new_for_msg` and `new_for_worker_set`, where `new_for_worker_set` doesn't take an index